tools/site_admin/manage_special_days.php

```diff
@@ -244,27 +244,27 @@ function show_edit_form()
                 <table class='edit_special_day'>";
             $this->_show_summary_row(_('Special Day ID'),$this->spec_code);
         }
-        $this->_show_edit_row('display_name',_('Display Name'),false,80);
+        $this->_show_edit_row('display_name', _('Display Name'), 'text', 80);
         echo "  <tr><th class='label'>Enable</th><td><input type='checkbox' name='enable'";
         if ( $this->enable )
             echo " value='1' checked";
         echo "></td></tr>\n";
-        $this->_show_edit_row('comment',_('Comment'),true);
-        $this->_show_edit_row('color',_('Color'),false,8);
-        $this->_show_edit_row('open_month',_('Open Month'),false,2);
-        $this->_show_edit_row('open_day',_('Open Day'),false,2);
-        $this->_show_edit_row('close_month',_('Close Month'),false,2);
-        $this->_show_edit_row('close_day',_('Close Day'),false,2);
-        $this->_show_edit_row('date_changes',_('Date Changes'),false);
-        $this->_show_edit_row('info_url',_('Info URL'),false);
-        $this->_show_edit_row('image_url',_('Image URL'),false);
+        $this->_show_edit_row('comment', _('Comment'), 'textarea');
+        $this->_show_edit_row('color', _('Color'), 'text', 8);
+        $this->_show_edit_row('open_month', _('Open Month'), 'number', null, 1, 12);
```

Comment:
Can't access this page, but it looks fine, might be a good idea to add `required` on them.